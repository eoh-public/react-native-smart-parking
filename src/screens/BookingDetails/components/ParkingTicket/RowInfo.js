import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import InfoField from 'components/SmartParking/BookingDetails/InfoField';

const RowInfo = memo(
  ({ leftValue, rightValue, leftTitle, rightTitle, body = true }) => {
    return (
      <View style={styles.container}>
        <InfoField title={leftTitle} value={leftValue} body={body} />
        <InfoField
          title={rightTitle}
          value={rightValue}
          style={styles.right}
          body={body}
        />
      </View>
    );
  }
);

export default RowInfo;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  text: {
    marginTop: 8,
  },
  right: {
    paddingLeft: 16,
  },
});
