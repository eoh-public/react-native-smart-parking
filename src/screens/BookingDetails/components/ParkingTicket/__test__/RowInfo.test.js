import React from 'react';
import renderer, { act } from 'react-test-renderer';
import RowInfo from '../RowInfo';

describe('Test RowInfo', () => {
  let tree;
  test('render row info item', () => {
    act(() => {
      tree = renderer.create(
        <RowInfo
          leftValue="Left value"
          rightValue="Right value"
          leftTitle="Left title"
          rightTitle="Right title"
        />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
