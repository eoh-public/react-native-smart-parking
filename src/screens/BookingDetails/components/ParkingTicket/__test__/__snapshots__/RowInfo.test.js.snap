// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RowInfo render row info item 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#8C8C8C",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      Left title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      Left value
          
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingLeft": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#8C8C8C",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      Right title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      Right value
          
    </Text>
  </View>
</View>
`;
